engine/src/main/java/io/camunda/zeebe/engine/state/deployment/FormVersionManager.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.deployment;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import java.util.Optional;
+import org.agrona.DirectBuffer;
+import org.agrona.collections.Object2ObjectHashMap;
+
+public class FormVersionManager {
```

Comment:
❌ I don't like the duplication you add in this PR. I think both the `FormVersionManager` and the `ProcessVersionManager` do the exact same thing. I'd propose:

1. Rename the `ProcesVersionManager` to `VersionManager`.
2. Add the ColumnFamily type as a constructor parameter.
3. Rename the `processIdKey` to `idKey`.
4. Rename the `ProcessVersionInfo` to `VersionInfo`.
5. Change the javadoc to not make it refer to a process.

This should give you a version manager that can be used by both forms and processes and you can get rid of the interface.

Reply:
I agree with you on merge `ProcessVersionManager` and `FormVersionManager`. It was initial idea, but I thought that you prefer to keep everything separated 😂 

I'm not agree, instead, with rename the `ProcessVersionInfo` to `VersionInfo`. Mainly because of [this](https://github.com/camunda/zeebe/blob/843952f50da3981673d98e08ef0ea124b479b6ad/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/ProcessVersionInfo.java#L20-L23)

Reply:
> I'm not agree, instead, with rename the ProcessVersionInfo to VersionInfo. Mainly because of [this](https://github.com/camunda/zeebe/blob/843952f50da3981673d98e08ef0ea124b479b6ad/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/ProcessVersionInfo.java#L20-L23)

It's only the name of the property which is a hardcoded string, which isn't exposed anywhere. The naming makes perfect sense when using the object. I'd argue the code duplication is worse.

Reply:
Fair enough. I'll refactor everything 💪